pkgs/by-name/bi/bisq/package.nix

```diff
@@ -0,0 +1,157 @@
+    '';
+
+  bisq-tor = writeScript "bisq-tor" ''
+    #! ${bash}/bin/bash
+
+    exec ${tor}/bin/tor "$@"
+  '';
+
+  # Alejandro Garcia's public key was obtained from
+  # https://bisq.network/pubkey/E222AA02.asc
+  publicKey = ./pubkey.asc;
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "bisq2";
+  version = "2.1.2";
+
+  src = fetchurl {
+    url = "https://github.com/bisq-network/${pname}/releases/download/v${version}/Bisq-${version}.deb";
+    sha256 = "0zgv70xlz3c9mrwmiaa1dgagbc441ppk2vrkgard8zjrvk8rg7va";
+
+    # Verify the upstream Debian package prior to extraction.
+    # See https://bisq.wiki/Downloading_and_installing#Verify_installer_file
+    # This ensures that a successful build of this Nix package requires the Debian
+    # package to pass verification.
+    nativeBuildInputs = [ gnupg ];
+    downloadToTemp = true;
+
+    postFetch = ''
+      pushd $(mktemp -d)
+      export GNUPGHOME=./gnupg
+      mkdir -m 700 -p $GNUPGHOME
+      ln -s $downloadedFile ./Bisq-${version}.deb
+      ln -s ${signature} ./signature.asc
+      gpg --import ${publicKey}
+      gpg --batch --verify signature.asc Bisq-${version}.deb
+      popd
+      mv $downloadedFile $out
+    '';
+  };
+
+  signature = fetchurl {
+    url = "https://github.com/bisq-network/${pname}/releases/download/v${version}/Bisq-${version}.deb.asc";
+    sha256 = "sha256-WZhI8RDmb7nQqpCQJM86vrp8qQNg+mvRVdSPcDqgzxE=";
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    dpkg
+    imagemagick
+    makeWrapper
+    zip
+    gnupg
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = pname;
+      icon = pname;
+      desktopName = "Bisq 2";
+      genericName = "Decentralized bitcoin exchange";
+      categories = [
+        "Network"
+        "P2P"
+      ];
+    })
+
+    (makeDesktopItem {
+      name = "${pname}-hidpi";
+      exec = "${pname}-hidpi";
+      icon = pname;
+      desktopName = "Bisq 2 (HiDPI)";
+      genericName = "Decentralized bitcoin exchange";
+      categories = [
+        "Network"
+        "P2P"
+      ];
+    })
+  ];
+
+  unpackPhase = ''
+    dpkg -x $src .
+  '';
+
+  buildPhase = ''
+    # Replace the Tor binary embedded in tor.jar (which is in the zip archive tor.zip)
+    # with the Tor binary from Nixpkgs.
+
+    cp ${bisq-tor} ./tor
+    zip tor.zip ./tor
+    zip opt/bisq2/lib/app/tor.jar tor.zip
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib $out/bin
+    cp -r opt/bisq2/lib/app $out/lib
+
+    install -D -m 777 ${bisq-launcher ""} $out/bin/${pname}
+    substituteAllInPlace $out/bin/${pname}
+
+    install -D -m 777 ${bisq-launcher "-Dglass.gtk.uiScale=2.0"} $out/bin/${pname}-hidpi
+    substituteAllInPlace $out/bin/${pname}-hidpi
+
+    for n in 16 24 32 48 64 96 128 256; do
+      size=$n"x"$n
+      magick convert opt/${pname}/lib/Bisq2.png -resize $size ${pname}.png
+      install -Dm644 -t $out/share/icons/hicolor/$size/apps ${pname}.png
```

Comment:
Please replace pname by the literal string https://github.com/NixOS/nixpkgs/issues/277994